src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -268,43 +250,44 @@ private sealed class FinalState : CompilationTrackerState
                 public readonly UnrootedSymbolSet UnrootedSymbolSet;
 
                 /// <summary>
-                /// The final compilation, with all references and source generators run. This is distinct from
-                /// <see cref="Compilation"/>, which in the <see cref="FinalState"/> case will be the compilation
-                /// before any source generators were ran. This ensures that a later invocation of the source generators
-                /// consumes <see cref="Compilation"/> which will avoid generators being ran a second time on a compilation that
-                /// already contains the output of other generators. If source generators are not active, this is equal to <see cref="Compilation"/>.
+                /// The final compilation, with all references and source generators run. This is distinct from <see
+                /// cref="Compilation"/>, which in the <see cref="FinalState"/> case will be the compilation before any
+                /// source generators were ran. This ensures that a later invocation of the source generators consumes
+                /// <see cref="Compilation"/> which will avoid generators being ran a second time on a compilation that
+                /// already contains the output of other generators. If source generators are not active, this is equal
+                /// to <see cref="Compilation"/>.
                 /// </summary>
-                public override ValueSource<Optional<Compilation>> FinalCompilationWithGeneratedDocuments { get; }
+                public override Compilation FinalCompilationWithGeneratedDocuments { get; }
 
                 private FinalState(
-                    ValueSource<Optional<Compilation>> finalCompilationSource,
-                    ValueSource<Optional<Compilation>> compilationWithoutGeneratedFilesSource,
+                    Compilation finalCompilation,
+                    Compilation compilationWithoutGeneratedFilesSource,
```

Comment:
> The only thing I'd request changed in this PR was the FinalState has duplicate parameters, the "source" can be removed since the compilation is also being passed directly at that point. Should be an easy fix.

Done.